test/testdata/namer/dynamic_constant.rb

```diff
@@ -5,4 +5,12 @@ def self.foo
     X = 10
 #   ^ error: dynamic constant assignment
```

Comment:
We noticed that some errors are capitalized and some are lower case as seen in [this](https://github.com/sorbet/sorbet/blob/master/third_party/parser/codegen/generate_diagnostics.cc#L49) file.  If there is a preferred way of doing this let me know.

Reply:
For some reason, all parser errors are lower case. I think this is an artifact of porting the whitequark parser to C++.

I don't know whether or not we can change those easily, because our whitequark parser test harness is set up to be able to automatically import and test existing whitequark tests.

All other errors should be uppercase.

It would be nice to add an ENFORCE here to maintain that invariant:

https://github.com/sorbet/sorbet/blob/df5101e2e955679a9e28b6f9bee6625d38fc81f9/core/Error.h#L102-L103